### DISCUSSION TITLE
Object graph comparison

### DISCUSSION THREAD

**Discussion by Greybird** [nunit/nunit#3955] [Ideas]
Hi,

Currently, it is sometimes cumbersome to create comparers, formatters, and to register them, to compare objects, or a subset of object properties.

Would you consider adding to Nunit assertion mechanism `Assert.That`, a way to express Object Graph comparison, in the way [FluentAssertions does for example](https://fluentassertions.com/objectgraphs/) ?

Thanks,

**Comment by mikkelbu** [comment]
Hi @Greybird 

Personally, I think we should leave these more advanced assertion features to FluentAssertions and Shouldly (and other assertion librarires), as we currently have a hard time finding (enough) time to work on the NUnit core functionality (definition and execution of tests). So I think we should focus our energy at this. 

Of course it is a bit unfortunate having tests where some of the assertions are written using NUnit syntax and other are written using FluentAssertion syntax, but perhaps we can bridge this gap to some degree without too much effort ?

**Reply by Greybird** [reply]
Hi @mikkelbu,

Thanks for your feedback.
I understand this position.

I'll try to invest a bit of time to assess how I could maybe wrap such features from one of these assertion libraries, to expose it in the NUnit Assert.That syntax ; this would probably be enough for my use case.

**Reply by Constante-ou-Mort** [reply]
@Greybird Hello, hope you're doing well, do we have any updates about it?

**Reply by Contagion21** [reply]
Welcome to 2025. FluentAssertions has gone to a paid license model so adding this functionality to Nunit would be welcomed by many

**Reply by stevenaw** [reply]
FWIW, @mikkelbu 's points about time and team focus are as valid today as they were then. That said, I think there are also a few more pieces in NUnit today that would make it simpler to add such a feature now, if the team chose to pursue it or accept a PR for it.

@Contagion21 have you considered also looking at Shouldly for a more immediate solution? I have found this PR which indicates it may have been added there as well: https://github.com/shouldly/shouldly/pull/411

**Reply by manfred-brands** [reply]
@Greybird

Does the [UsingPropertiesComparer](https://docs.nunit.org/articles/nunit/writing-tests/constraints/EqualConstraint.html#comparing-classesstructures) option help you to compare object graphs?
There are two updates pending:
1. Comparing of a subset of properties #4687 
2. Comparing matching properties on different types #4935 

The PR is  #4925 

Is there anything missing from that PR you need to compare your objects?
